apisix/core/config_etcd.lua

```diff
@@ -257,6 +260,33 @@ local function run_watch(premature)
 end
 
 
+local function run_watch(premature)
+    local run_watch_th = ngx_thread_spawn(do_run_watch, premature)
+
+    ::restart::
+    local check_worker_th = ngx_thread_spawn(function ()
+        while not exiting() do
+            ngx_sleep(0.1)
+        end
+    end)
+
+    local ok, err = ngx_thread_wait(check_worker_th)
+
+    if not ok then
+        log.error("wait worker checker failed, retart checker, error: " .. err)
+        ngx_thread_kill(check_worker_th)
+        goto restart
+    end
+
+    -- waiting for do_run_watch thread to handle the final etcd event
+    ngx_sleep(1)
```

Comment:
How do you know no more etcd events will come in the last 1 second?
I think you should check ngx.exiting in `do_run_watch` too

Reply:
After we execute `make quit` command, the admin API will not be available, so before the admin API becomes unavailable, we can use`sleep(1)` to wait for the `do_run_watch` thread to handle the last possible event. Even if no events come, we also will kill the `do_run_watch` thread. 
And, check `ngx.exiting` in `do_run_watch` it means we need to wait for the `res_func()` function timeout,  that's why I create two threads to handle it.

Reply:
The DP don't have admin API, it use etcd to sync the configuration from the CP which has admin API, when you execute `make quit` on DP, the CP is available.
So you should stop syncing after you execute `make quit` on DP

Reply:
By the way, do we really need to do this before quit? we will load full data after reload

Reply:
Yes, it is not necessary. `t/node/healthcheck-stop-checker.t.` this test case failed seems related to this,  so I add the logic.

Reply:
I will try it.